### DISCUSSION TITLE
phase and all-phases duration

### DISCUSSION THREAD

**Discussion by pawel-borkowski** [artilleryio/artillery#1754] [Artillery]
How can I get the duration time of a phase and duration of all phaseses after a test run. I can not find them in the report.json file.

**Comment by juanig1** [comment]
Hey!
Report only aggregates metrics for each 10s period (as outputted by the CLI when running the test), so:
- not possible right now to get phases and their duration only from the report file
- You still can get the total test duration using the `aggregate.firstMetricAt` and `aggregate.lastMetricAt`. [This](https://github.com/artilleryio/artillery/blob/main/packages/artillery/lib/report/index.html.ejs#L541-L555) is how we handle it right now in our report logic

**Comment by pawel-borkowski** [comment]
Hey,
Thanks!
When `aggregate.firstMetricAt` and `aggregate.lastMetricAt` are taken?

**Reply by pawel-borkowski** [reply]
I will rephrase: in what moment of test execution `aggregate.firstMetricAt` and `aggregate.lastMetricAt` are collected?

**Reply by juanig1** [reply]
Sorry, missed the first message.

Each time a VU is generated we generate these stats (period, statuscode, etc). After execution all metrics are aggregated and from there we get the first and last timestamp to get `firstMetricAt` and `lastMetricAt` respectively. 

This is not strictly the duration of the phase but the duration from test start until the last VU was created (which may be important to you!).

Some alternatives to get the full duration I can thing of:
- If you can access the original script you may be able to `grep` durations from it
- You could add a custom js handler to print/save/etc phases durations
- `artillery run my.yaml | grep 'Phase completed' > durations.txt` may be helpful too (you may have to enrich the grep pattern a bit to get all the info you need)
